CHANGELOG.md

```diff
@@ -1,3 +1,75 @@
+* Adding multi-package support for pubrootdirectories [#4306](https://github.com/flutter/devtools/pull/4306)
+* Refactor RequestableSizeWidget [#4340](https://github.com/flutter/devtools/pull/4340)
+* Use G3 compatible icons [#4347](https://github.com/flutter/devtools/pull/4347)
+* App Size panel clean-up [#4346](https://github.com/flutter/devtools/pull/4346)
+* Adding ArtificialRoot to non-deferred apps for diff table [#4338](https://github.com/flutter/devtools/pull/4338)
+* Expand and Unexpand top nodes in Treetable [#4313](https://github.com/flutter/devtools/pull/4313)
+* Update release_notes README.md [#4345](https://github.com/flutter/devtools/pull/4345)
+* Update broken link on release instructions [#4344](https://github.com/flutter/devtools/pull/4344)
+* Field Display initial implementation [#4322](https://github.com/flutter/devtools/pull/4322)
+* Bump github/codeql-action from 2.1.17 to 2.1.18 [#4341](https://github.com/flutter/devtools/pull/4341)
+* Turn on Frame Analysis feature by default and add more analytics [#4339](https://github.com/flutter/devtools/pull/4339)
+* Move framework-only code to the folder 'framework' [#4334](https://github.com/flutter/devtools/pull/4334)
+* Write tests for frame hints [#4337](https://github.com/flutter/devtools/pull/4337)
+* Migrate ClipboardStatusNotifier [#4122](https://github.com/flutter/devtools/pull/4122)
+* Add a dropdown to App Size analysis tab [#4332](https://github.com/flutter/devtools/pull/4332)
+* Forward args from devtools build_e2e.dart to pkg:dds/serve_local.dart [#4331](https://github.com/flutter/devtools/pull/4331)
+* Add key and value types to VmInfoCard mapEntries [#4329](https://github.com/flutter/devtools/pull/4329)
+* Extract some span_parser tests into "golden" tests [#4325](https://github.com/flutter/devtools/pull/4325)
+* Rewrite FrameTimeVisualizer in performance page [#4328](https://github.com/flutter/devtools/pull/4328)
+* Fix layoutPhase calculation [#4327](https://github.com/flutter/devtools/pull/4327)
+* Add information about performance impact [#4324](https://github.com/flutter/devtools/pull/4324)
+* Adding deferred indicator on the treetable [#4305](https://github.com/flutter/devtools/pull/4305)
+* Update LEAK_TRACKING.md [#4273](https://github.com/flutter/devtools/pull/4273)
+* Update release instructions [#4320](https://github.com/flutter/devtools/pull/4320)
+* Bump to latest flutter beta [#4319](https://github.com/flutter/devtools/pull/4319)
+* Minor changes for leak tracking doc [#4318](https://github.com/flutter/devtools/pull/4318)
+* Bump github/codeql-action from 2.1.15 to 2.1.17 [#4317](https://github.com/flutter/devtools/pull/4317)
+* Add MacOS bot to the CI [#4078](https://github.com/flutter/devtools/pull/4078)
+* Improve helper functions in vm_developer_common_widgets [#4312](https://github.com/flutter/devtools/pull/4312)
+
+## 2.17.0
+TODO: update changelog
+
 ## 2.16.0-dev.0
```

Comment:
okay a couple things here. 
1. Not due to this PR, but this should have been 2.17.0-dev.0 since 2.16.0-dev.0 would technically be before 2.16.0.
2. We should squash any dev version Changelog entries into the non-dev major / minor version once it is ready. So all commits between 2.16.0 and 2.17.0 should be under the 2.17.0 entry and the dev entry should be removed. In general, we can omit changelog entries for dev releases, and should probably change the update_version script to not add the Changelog TODO for `--type dev`.

CC @CoderDake for visibility

Reply:
squashed